core/src/impl/Kokkos_Default_GraphNodeKernel.hpp

```diff
@@ -36,15 +36,22 @@ struct GraphNodeKernelDefaultImpl {
   // TODO @graphs decide if this should use vtable or intrusive erasure via
   //      function pointers like in the rest of the graph interface
   virtual void execute_kernel() = 0;
+
+  GraphNodeKernelDefaultImpl() = default;
+
+  GraphNodeKernelDefaultImpl(ExecutionSpace exec)
+      : m_execution_space(std::move(exec)) {}
+
+  ExecutionSpace m_execution_space;
 };
 
 // TODO Indicate that this kernel specialization is only for the Host somehow?
 template <class ExecutionSpace, class PolicyType, class Functor,
           class PatternTag, class... Args>
 class GraphNodeKernelImpl
-    : public PatternImplSpecializationFromTag<PatternTag, Functor, PolicyType,
-                                              Args..., ExecutionSpace>::type,
-      public GraphNodeKernelDefaultImpl<ExecutionSpace> {
+    : public GraphNodeKernelDefaultImpl<ExecutionSpace>,
+      public PatternImplSpecializationFromTag<PatternTag, Functor, PolicyType,
+                                              Args..., ExecutionSpace>::type {
```

Comment:
Because I'm extracting the execution space instance from the execution policy to pass it to the constructor of `GraphNodeKernelDefaultImpl`. Then, the policy is moved to the constructor of `PatternImplSpecializationFromTag`.

Or maybe I'm wrong.